File: 05_salary_increase/ocaml/bin/main.ml
let increment_percentage = 0.1

let () = print_endline "\n--- PROGRAMA PARA AUMENTAR EL SALARIO ---\n"

let () = print_endline "Ingresa tu salario actual:"
let salary = float_of_string (read_line ())

let () = print_endline "Ingresa el salario base:"
let base = float_of_string (read_line ())

let () = if salary < 2.0 *. base then
  let increase = salary *. increment_percentage in
  let new_salary = salary +. increase in
  let () = print_endline (Printf.sprintf "\nTu aumento será de: $ %f" increase) in
  print_endline (Printf.sprintf "Tu nuevo salario será de: $ %f" new_salary)
else
  print_endline "\nNo tienes derecho a un aumento"

File: 05_salary_increase/rust/src/main.rs
use std::io;
use std::io::Write;

const INCREMENT_PERCENTAGE: f32 = 0.1;

fn main() {
    let mut input = String::new();

    println!("\n--- PROGRAMA PARA AUMENTAR EL SALARIO ---\n");

    print!("Ingresa tu salario actual > ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut input).unwrap();
    let salary = input.trim().parse::<f32>().expect("El salario debe ser un número");

    print!("Ingresa el salario base > ");
    io::stdout().flush().unwrap();
    input.clear();
    io::stdin().read_line(&mut input).unwrap();
    let base = input.trim().parse::<f32>().expect("El salario debe ser un número");

    if salary < 2.0 * base {
        let increase = salary * INCREMENT_PERCENTAGE;
        let new_salary = salary + increase;

        println!("\nTu aumento será de: $ {increase}");
        println!("Tu nuevo salario será de: $ {new_salary}");
    } else {
        println!("\nNo tienes derecho a un aumento");
    }
}
